codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ServiceClientConfigurationUtils.java

```diff
@@ -81,26 +86,54 @@ public List<Field> serviceClientConfigurationFields() {
         return Collections.unmodifiableList(fields);
     }
 
-    private List<Field> fields() {
-        return Arrays.asList(
+    private List<Field> fields(IntermediateModel model) {
+        List<Field> fields = new ArrayList<>();
+
+        fields.addAll(Arrays.asList(
             overrideConfigurationField(),
             endpointOverrideField(),
             endpointProviderField(),
             regionField(),
             credentialsProviderField(),
             authSchemesField(),
             authSchemeProviderField()
-        );
+        ));
+        fields.addAll(addCustomClientParams(model));
+        return fields;
+    }
+
+    private List<Field> addCustomClientParams(IntermediateModel model) {
+        List<Field> customClientParamFields = new ArrayList<>();
+
+        if (model.getCustomizationConfig() != null && model.getCustomizationConfig().getCustomClientContextParams() != null) {
+            model.getCustomizationConfig().getCustomClientContextParams().forEach((n, m) -> {
+
+                String paramName = endpointRulesSpecUtils.paramMethodName(n);
+                String keyName = model.getNamingStrategy().getEnumValueName(n);
+                TypeName type = endpointRulesSpecUtils.toJavaType(m.getType());
+
+                customClientParamFields.add(fieldBuilder(paramName, type)
+                      .doc(m.getDocumentation())
+                      .isInherited(false)
+                      .localSetter(basicLocalSetterCode(paramName))
+                      .localGetter(basicLocalGetterCode(paramName))
+                      .configSetter(customClientConfigParamSetter(paramName, keyName))
+                      .configGetter(customClientConfigParamGetter(keyName))
+                      .build());
+            });
+        }
+
+        return customClientParamFields;
     }
 
     private Field overrideConfigurationField() {
         return fieldBuilder("overrideConfiguration", ClientOverrideConfiguration.class)
-                    .doc("client override configuration")
```

Comment:
Just indentation. Let me know if we prefer to let these be, I can see that it might complicate the PR